### PR TITLE
fix(ui): Correct button icon sizing / padding

### PR DESCRIPTION
**Pull request by evanpurkhiser**
@chrissy took this pull over and henceforth owns whatever mess it makes.

----

We had some crazy stuff going on in `button.jsx`:

1. we were applying a negative margin of 2px to all component icons
2. we were applying a positive margin of 2px to all child elements with a `.icon` class
3. we are using a lot of `span` elements inside of `button`, so `button` thinks it has a label instead of an icon

`button` shouldn't try to do anything if you want to inject a span with an icon font attached to it. If you are gonna do it wrong, then at least style it yourself.

With that in mind, I removed all of the crazy margin stuff, and changed a bunch of the icon-font-based trash icons to use the new `icon` prop. much cleaner!

I visually verified every change that was brought to my attention via the snapshot tests and will do the same obviously for percy. tbh we really only use three icons in our buttons: trash, plus, and minus. It was simpler than I thought.

### CODE REVIEW

**Review by Chrissy — DISMISSED** [review]

**Comment by EvanPurkhiser** [comment]
Having @Chrissy take this one over

**Comment by Chrissy** [comment]
@ckj I assigned you to this, since we discussed the general approach yesterday.

I approved it two days ago during simpler times when it was Evan's pull. I tried to remove the approval but it won't let me :|

**Comment by Chrissy** [comment]
wow @EvanPurkhiser that was so pro how did you do that?!

**Comment by EvanPurkhiser** [comment]
@Chrissy I think you just have to own the PR to do it. It was under this expanded panel:

image

File: src/sentry/conf/server.py [hunk 1]
```diff
@@ -753,12 +753,12 @@ def create_partitioned_queues(name):
     'organizations:sso-saml2': False,
     'organizations:sso-rippling': False,
     'organizations:group-unmerge': False,
-    'organizations:github-apps': True,
-    'organizations:integrations-v3': True,
+    'organizations:github-apps': False,
```
Comment (ckj) [inline src/sentry/conf/server.py]: Did you mean to commit these changes?

Comment (ckj) [reply src/sentry/conf/server.py]: Oh, okay. Weird, not sure how I ended up seeing them. Carry on!

Comment (Chrissy) [reply src/sentry/conf/server.py]: yeah sorry about that!

**Comment by EvanPurkhiser** [comment]
Can't add review, but this LGTM

**Review by ckj — APPROVED** [review]
🤘

**Pull request merged**